### PR TITLE
update Torque documentation

### PR DESCRIPTION
**Pull request by tebbi**

### CODE REVIEW

File: src/_img/docs/torque/build-process.svg [hunk 1]
```diff
@@ -1 +1,491 @@
...
+       fill="#fbc02d"
+       id="path91" />
+    <path
+       d="m49.498 387.772h-2.266v4.64h-1.437v-11.375h3.328q.797.016 1.5.235.703.203 1.234.625.531.421.828 1.062.297.625.297 1.484 0 .563-.156 1.032-.156.453-.438.828-.28.375-.672.672-.39.281-.859.484l2.406 4.86v.093h-1.53zm-2.266-1.188h1.938q.484-.016.906-.156.437-.14.75-.406.328-.282.515-.672.188-.407.188-.922 0-.547-.188-.953-.171-.407-.5-.688-.312-.281-.75-.422-.437-.14-.968-.14h
```
Comment (mathiasbynens) [inline src/_img/docs/torque/build-process.svg:491]: Please run this through [SVGOMG](https://jakearchibald.github.io/svgomg/).

File: src/docs/torque.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ Torque provides language constructs to represent high-level, semantically-rich t
...
 
-Most source written in Torque is checked into the V8 repository under [the `src/builtins` directory](https://github.com/v8/v8/tree/master/src/builtins), with the file extension `.tq`. (The actual Torque compiler can be found under [`src/torque`](https://github.com/v8/v8/tree/master/src/torque).). Tests for Torque functionality are checked in under [`test/torque`](https://github.com/v8/v8/tree/master/test/torque).
+Most source written in Torque is checked into the V8 repository under [the `src/builtins` directory](https://github.com/v8/v8/tree/master/src/builtins), with the file extension `.tq`. Torque definitions of V8's heap-allocated classses are found alongside their C++ definitions, in `.tq` files with the same name as corresponding C++ files in `src/objects`.  The actual Torque compiler can be found under [`src/torque`](https://github.com/v8/v8/tree/master/src/torque). Tests for Torque functionality are checked in under [`test/torque`](https://github.com/v8/v8/tree/master/test/torque), [`test/cctest/torque`](https://github.com/v8/v8/tree/master/test/cctest/torque), and [`test/unittests/torque`](https://github.com/v8/v8/tree/master/test/unittests/torque).
```
Comment (mathiasbynens) [inline src/docs/torque.md:13]:
```suggestion
Most source written in Torque is checked into the V8 repository under [the `src/builtins` directory](https://github.com/v8/v8/tree/master/src/builtins), with the file extension `.tq`. Torque definitions of V8's heap-allocated classses are found alongside their C++ definitions, in `.tq` files with the same name as corresponding C++ files in `src/objects`. The actual Torque compiler can be found under [`src/torque`](https://github.com/v8/v8/tree/master/src/torque). Tests for Torque functionality are checked in under [`test/torque`](https://github.com/v8/v8/tree/master/test/torque), [`test/cctest/torque`](https://github.com/v8/v8/tree/master/test/cctest/torque), and [`test/unittests/torque`](https://github.com/v8/v8/tree/master/test/unittests/torque).
```

File: src/docs/torque.md [hunk 3]
```diff
@@ -94,29 +97,12 @@ In combination with generics, `constexpr` is a powerful Torque tool that can be
...
-
-NamespaceDeclaration :
-  namespace IdentifierName { Declaration* }
-
-FileDeclaration :
-  NamespaceDeclaration
-  Declaration
-```
+Torque code is packaged in individual source files. Each source file consists of a series of declarations, which themselves can optionally wrapped in a namespace declaration to separate the namespaces of declarations. The following description of the grammar is likely out-of-date. The source-of-truth is [the grammar definition in the Torque compiler](https://source.chromium.org/chromium/chromium/src/+/master:v8/src/torque/torque-parser.cc?q=TorqueGrammar::TorqueGrammar), which is written using contex-free grammar rules.
+A Torque file is a sequence of declarations. The possible declarations are listed [here](https://source.chromium.org/chromium/chromium/src/+/master:v8/src/torque/torque-parser.cc?q=TorqueGrammar::declaration).
```
Comment (mathiasbynens) [inline src/docs/torque.md:101]:
```suggestion

A Torque file is a sequence of declarations. The possible declarations are listed [in `torque-parser.cc`](https://source.chromium.org/chromium/chromium/src/+/master:v8/src/torque/torque-parser.cc?q=TorqueGrammar::declaration).
```

**Comment by tebbi** [comment]
I addressed the comments, PTAL again.

**Review by mathiasbynens — APPROVED** [review]
Woah! Blast from the past :)

File: src/docs/torque.md [hunk 4]
```diff
@@ -595,7 +587,8 @@ IntrinsicDeclaration :
   intrinsic % IdentifierName ImplicitParameters opt ExplicitParameters ReturnType opt ;
 ```
 
-For the most part, “user” Torque code should rarely have to use `intrinsic`s directly. The currently supported intrinsics are:
+For the most part, “user” Torque code should rarely have to use `intrinsic`s directly.
+The following are some of the  supported intrinsics:
```
Comment (mathiasbynens) [inline src/docs/torque.md:591]:
```suggestion
The following are some of the supported intrinsics:
```

**Pull request merged**